source/extensions/filters/listener/local_ratelimit/local_ratelimit.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/extensions/filters/listener/local_ratelimit/v3/local_ratelimit.pb.h"
+#include "envoy/network/filter.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/runtime/runtime_protos.h"
+#include "source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace LocalRateLimit {
+
+/**
+ * All local rate limit stats. @see stats_macros.h
+ */
+#define ALL_LOCAL_RATE_LIMIT_STATS(COUNTER) COUNTER(rate_limited)
+
+/**
+ * Struct definition for all local rate limit stats. @see stats_macros.h
+ */
+struct LocalRateLimitStats {
+  ALL_LOCAL_RATE_LIMIT_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+using LocalRateLimiterImplSharedPtr =
+    std::shared_ptr<Filters::Common::LocalRateLimit::LocalRateLimiterImpl>;
+
+/**
+ * Configuration shared across all connections. Must be thread safe.
+ */
+class Config : Logger::Loggable<Logger::Id::filter> {
+public:
+  Config(
+      const envoy::extensions::filters::listener::local_ratelimit::v3::LocalRateLimit& proto_config,
+      Event::Dispatcher& dispatcher, Stats::Scope& scope, Runtime::Loader& runtime);
+
+  ~Config() = default;
+
+  bool canCreateConnection();
+  bool enabled() { return enabled_.enabled(); }
+  const LocalRateLimitStats& stats() const { return stats_; }
+
+private:
+  static LocalRateLimitStats generateStats(const std::string& prefix, Stats::Scope& scope);
+
+  Runtime::FeatureFlag enabled_;
+  LocalRateLimitStats stats_;
+  LocalRateLimiterImplSharedPtr rate_limiter_;
```

Comment:
make sense. thanks! i'll fix network/local_ratelimit in a separated PR

Reply:
I am not sure if we need to fix `network/local_ratelimit`. 

My understanding of `shared_ptr` there is 1) if `share_key` has value(i.e. sharing is enabled), they leverage shared_ptr 2) if `share_key` is empty(i.e., sharing is not enable), they just `make_shared` every time it is created, so each one has separate control with ref_count=1, which basically use it like a unique_ptr. 
And I think the reason `shared_ptr` is used there is because of coding convenience/consistence as you just need same type of object for above two cases.

Thank you for addressing my comments and adding tests! I will take a look later.